### ISSUE TITLE
Group confusion

### ISSUE THREAD

**Issue by jimtng** [boc-tothefuture/openhab-jruby#47]
I would normally expect that when I refer to GroupItem1 it would return me the group's state. But instead, it seems to resolve to an array of member items.

This is where it tripped me:
```
rule 'xx' do
  changed xxx
  only_if { GroupItem1 == OPEN } # This would fail regardless of GroupItem1's state
...
```

Currently I'd have to write it as:
```
only_if { GroupItem1.group == OPEN }
```

I'd like to propose that GroupItem1 would return the state of the group itself, not its item members.

**Comment by jimtng** [comment]
Note if this behaviour gets changed, we need to update store_states implementation. It wouldn't have to treat Group as a special case.

**Comment by jimtng** [comment]
Digging further, this part also confused me, but I got clarifications from the source code

Given: 
```
Group Group1
Switch Item1 (Group1)
Switch Item2 (Group1)
```

`Group1.items`

At a glance, this appears to give me the members of the group, in other words, I'd expect that Group1.items is an array containing [ Item1, Item2 ]. But this is not the case. Group1.item is merely a wrapper for the OpenHAB's GroupItem, for the sole purpose of feeding it to the triggers e.g. `changed`, `updated` or `received_command`, so that they will use "Member Of" trigger.

In other words:

---
```ruby
rule 'x' do
  changed Group1.items
end
```
is equivalent to RulesDSL
```
rule "x"
when
  Member of Group1 changed
then
end
```
---

Versus:
---
```ruby
rule 'x' do
  changed Group1
end

#Or equivalently:
rule 'x' do
  changed Group1.group # exactly the same result as above
end
```
is equivalent to RulesDSL
```
rule "x"
when
  Item Group1 changed
then
end
```
---


If this is correct, at the very least I think it should be explained further in the documentation / readme.

**Comment by boc-tothefuture** [comment]
So groups are weird because you want to be able to trigger on the group itself, or trigger on the items in the group.  In the original design I also wanted to be able to operate directly on groups with enumerable methods so you could grep, select, any?, all?, max, min, etc and have that operate on the items.

It seemed that putting 'items' in front of all that was unnecessary, certainly when you consider a group is a set of items so you should be able to just operate on them naturally.  

If we increase the documentation and if we change == to compare the states do we solve the problem or do we think we have a broader issue?

**Comment by jimtng** [comment]
This is related to #48 too. Ideally, I'd like all the item types (StringItem, NumberItem, SwitchItem, Group (yes it's an item), ContactItem, etc) to be of Java type. 

In the case of GroupItem wanting to trigger on the group itself, it should just be `changed GroupItem` without any special handling that is done now. For "Member of" type trigger, we could settle on GroupItem.items as it is now.
